xonsh/readline_shell.py

```diff
@@ -281,6 +282,27 @@ def inserter():
     return inserter
 
 
+def _render_completions(completions, prefix, prefix_len):
+    """Render the completions according to the required prefix_len.
+
+    Readline will replace the current prefix with the chosen rendered completion.
+    """
+    chopped = prefix[:-prefix_len] if prefix_len else prefix
+
+    rendered_completions = []
+    for comp in completions:
+        if isinstance(comp, RichCompletion) and comp.prefix_len is not None:
+            if comp.prefix_len == 0:
```

Comment:
Should be `if comp.prefix_len:` for speed (and switch conditions)

Reply:
done

Reply:
done